### ISSUE TITLE
[documentation] clarify differences between grpc, swagger and conjure

### ISSUE THREAD

**Issue by pwilczynski** [palantir/conjure#245]
In this PR (https://github.com/palantir/conjure/pull/16) , @iamdanfox mentioned adding documentation to help understand more of how to model conjure in relation to a few adjacent projects including protobuf, grpc, swagger and OpenAPI.

Ideally this documentation would split out the concepts of the interface definition language, the transport / serialization layer and language-specific bindings and explain where conjure fits.

**Comment by pwilczynski** [comment]
Also Avro, Thrift, etc

**Comment by carterkozak** [comment]
Some of this is answered here:
https://medium.com/palantir/introducing-conjure-palantirs-toolchain-for-http-json-apis-2175ec172d32

@uschi2000 Given your early involvement with conjure, would you be interested in adding to or expanding upon this? It would be helpful if this repository provided that documentation, minimally a link to the blog post.

**Comment by uschi2000** [comment]
Yep, can do.

**Comment by pwilczynski** [comment]
Yeah, the blog post is really helpful, but I'm also curious about a 2020 assessment of how other technologies have evolved since we started working on Conjure in 2016/2017, especially the newer support of GRPC for typescript and the evolution of swagger.

In particular, curious about how you model the relationship between this and some of the other OSS tools that are evolving.
